### PR TITLE
Support quoting values in where clauses

### PR DESCRIPTION
**Pull request by leondutoit**
Addresses: https://github.com/unioslo/pysquril/issues/38

Library functionality:
* quote values in where clauses - allows using previously reserved words and characters in searches

Internal changes:
* parser supports quoting
* new tests for parsing

### CODE REVIEW

File: pysquril/tests.py [hunk 1]
```diff
@@ -274,6 +306,16 @@ def run_delete_query(
...
         out = run_select_query('select=z&where=float=gt.3.2')
         assert out[0][0] == 1
+        # with quoting
+        out = run_select_query("select=x&where=meh=eq.'.'")
+        assert out[0][0] == 10
+        out = run_select_query("select=x&where=lolly=eq.'()'")
+        assert out[0][0] == 10
+        out = run_select_query("select=x&where=wat=eq.'and:'")
+        assert out[0][0] == 10
+        out = run_select_query("select=x&where=meh2=eq.'()[],and:,or:. where=;'")
```
Comment (kouylekov-usit) [inline pysquril/tests.py:316]: Nice

**Review by kouylekov-usit — APPROVED** [review]

File: pysquril/parser.py [hunk 2]
```diff
@@ -151,39 +151,62 @@ def categorise_groups(self, groups: list) -> tuple:
...
-        for char in self.original:
-            if char in ['(', ')']:
-                groups.append(char)
-        element = element.replace('(', '')
-        element = element.replace(')', '')
-        combinators = ['and:', 'or:']
+        # find groups, remove brackets
+        is_quoted = False
+        for token in element:
+            if token == "'":
```
Comment (egiltane) [inline pysquril/parser.py:167]: Does that mean that any evenly numbered amount of single quotes switches the quoting back off? If so, is that a problem?

File: pysquril/parser.py [hunk 3]
```diff
@@ -255,26 +278,29 @@ def __init__(self, original: str) -> None:
...
-            if token == ',' and brace_open and not brace_closed:
-                token = ';'
-            temp += token
-        parts = temp.split(',')
-        for part in parts:
-            if ';' in part:
-                part = part.replace(';', ',')
-            out.append(part)
-        return out
+            if token == "'":
```
Comment (egiltane) [inline pysquril/parser.py:286]: Same here.

**Review by egiltane — APPROVED** [review]
Nice, indeed.

File: pysquril/parser.py (hunk 2 above)
Comment (leondutoit) [reply pysquril/parser.py:167]: It is supposed to work like that, first `'`, quoting starts, second `'`, quoting has ended. Until there is support for escaping quotes inside the quotes by using escape characters such as `\'`, but I left that out for now.

**Pull request merged**